src/test/cypress/integration/exam/ExamAssessment.spec.ts

```diff
@@ -0,0 +1,186 @@
+            cy.get('.question-options').contains('7 of 10 points').should('be.visible');
+        });
+    });
+
+    describe('Exam Programming Exercise Assessment', () => {
+        let programmingExerciseName: string;
+        let programmingExerciseShortName: string;
+        const examEnd = (Cypress.env('isBamboo') ? GROUP_SYNCHRONIZATION / 1000 : 0) + 115;
+
+        before('Generate exercise names', () => {
+            uid = generateUUID();
+            programmingExerciseName = 'Cypress programming exercise ' + uid;
+            programmingExerciseShortName = 'cypress' + uid;
+        });
+
+        beforeEach('Create Exam', () => {
+            prepareExam(dayjs().add(examEnd, 'seconds'));
+        });
+
+        it('Assess a programming exercise submission (MANUAL)', () => {
+            cy.login(admin);
+            courseManagementRequests.createProgrammingExercise({ exerciseGroup }, programmingExerciseName, programmingExerciseShortName, packageName).then((progRespone) => {
+                const programmingExercise = progRespone.body;
+                courseManagementRequests.generateMissingIndividualExams(exam);
+                courseManagementRequests.prepareExerciseStartForExam(exam);
+                cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                examStartEnd.startExam();
+                cy.contains(programmingExercise.title).should('be.visible').click();
+                makeSubmissionAndVerifyResults(partiallySuccessful, () => {
+                    cy.get('#exam-navigation-bar').find('.btn-danger').click();
+                    examStartEnd.finishExam();
+                    cy.get('.alert').should('be.visible');
+                    cy.login(tutor, '/course-management/' + course.id + '/exams');
+                    cy.contains('Assessment Dashboard', { timeout: examEnd * 1000 }).click();
+                    assessmentDashboard.startAssessing();
+                    assessmentDashboard.addNewFeedback(2, 'Good job');
+                    assessmentDashboard.submitAssessment();
+                    cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
+                    cy.get('.question-options').contains('6.6 of 10 points').should('be.visible');
+                });
+            });
+        });
+    });
+});
+
+function makeSubmissionAndVerifyResults(submission: ProgrammingExerciseSubmission, verifyOutput: () => void) {
+    // We create an empty file so that the file browser does not create an extra subfolder when all files are deleted
+    editorPage.createFileInRootPackage('placeholderFile');
+    // We delete all existing files, so we can create new files and don't have to delete their already existing content
+    editorPage.deleteFile('Client.java');
+    editorPage.deleteFile('BubbleSort.java');
+    editorPage.deleteFile('MergeSort.java');
+    editorPage.typeSubmission(submission, packageName);
+    editorPage.submit();
+    verifyOutput();
+}
```

Comment:
This function is copied from the ProgrammingExerciseParticipation.spec. Instead I suggest sharing the function. In my static code analysis PR I did exactly that by moving the [function into the online editor pageobject](https://github.com/ls1intum/Artemis/blob/2647f2d70a81ac9b973e48e3f554d68d3e2c3072/src/test/cypress/support/pageobjects/OnlineEditorPage.ts#L152)

Reply:
will be addressed in a future PR